src/test/java/org/cactoos/io/ReaderOfTest.java

```diff
@@ -40,18 +41,317 @@
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputWithCharset() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputWithCharsetByName() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8.name()
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputWithCharsetDecoder() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8.newDecoder()
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputStream() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputStreamOf("Hello, товарищ!")
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputStreamWithCharset() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputStreamOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputStreamWithCharsetByName() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputStreamOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8.name()
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
+
+    @Test
+    public void readsInputStreamWithCharsetDecoder() throws IOException {
+        MatcherAssert.assertThat(
+            "Can't read content",
+            new InputOf(
+                new ReaderOf(
+                    new InputStreamOf("Hello, товарищ!"),
+                    StandardCharsets.UTF_8.newDecoder()
+                )
+            ),
+            new InputHasContent("Hello, товарищ!")
+        );
+    }
 }
```

Comment:
@proshin-roman Can you also add a test for german characters? It is also quite a spoken language, so we should be able to read german diacritics. The german diacritics are: ``äÄ üÜ öÖ and ß``

Reply:
That was done too.